core/jobs/src/main/java/com/cognifide/aet/job/common/utils/ParamsHelper.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Automated Exploratory Tests
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.aet.job.common.utils;
+
+
+import com.cognifide.aet.job.api.exceptions.ParametersException;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public class ParamsHelper {
+
+  /***
+   * @param key property name that should store regexp
+   * @param params map of parameters
+   * @return Pattern  or null if there is no key in params map
+   * @throws ParametersException if Pattern under key is provided but invalid
+   */
+  public static Pattern getAsPattern(String key, Map<String, String> params) throws ParametersException {
```

Comment:
How about one of the the ideas for all this helper methods:

1. Don't return null, use Optional instead.
2. Don't return null, provide option with 3rd - default parameter and return it when no param with key provided.